source/common/http/http2/codec_impl.h

```diff
@@ -501,12 +513,12 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
   int onMetadataFrameComplete(int32_t stream_id, bool end_metadata);
   ssize_t packMetadata(int32_t stream_id, uint8_t* buf, size_t len);
   // Adds buffer fragment for a new outbound frame to the supplied Buffer::OwnedImpl.
-  // Returns true on success or false if outbound queue limits were exceeded.
-  bool addOutboundFrameFragment(Buffer::OwnedImpl& output, const uint8_t* data, size_t length);
-  virtual void checkOutboundQueueLimits() PURE;
-  void incrementOutboundFrameCount(bool is_outbound_flood_monitored_control_frame);
-  virtual bool trackInboundFrames(const nghttp2_frame_hd* hd, uint32_t padding_length) PURE;
-  virtual bool checkInboundFrameLimits(int32_t stream_id) PURE;
+  // Returns Ok Status on success or error if outbound queue limits were exceeded.
+  Status addOutboundFrameFragment(Buffer::OwnedImpl& output, const uint8_t* data, size_t length);
+  virtual Status checkOutboundQueueLimits() PURE;
+  Status incrementOutboundFrameCount(bool is_outbound_flood_monitored_control_frame);
```

Comment:
What do we think of ABSL_MUST_USE_RESULT for these to make it clear (and compiler checked ) we're doing the right thing?

Reply:
The whole `Status` class is marked like this [here](https://github.com/abseil/abseil-cpp/blob/41a6263fd0bcc93a90ff739785f17260f8ea061e/absl/status/status.h#L75), so any time a method returns Status it is marked that it must be checked.